crates/turborepo-auth/src/auth/login.rs

```diff
@@ -20,13 +32,45 @@ pub async fn login<'a>(
     existing_token: Option<&'a str>,
     login_url_configuration: &str,
     login_server: &impl LoginServer,
-) -> Result<Cow<'a, str>, Error> {
-    // Check if token exists first.
+) -> Result<Token, Error> {
+    // Check if passed in token exists first.
     if let Some(token) = existing_token {
         if let Ok(response) = api_client.get_user(token).await {
             println!("{}", ui.apply(BOLD.apply_to("Existing token found!")));
             ui::print_cli_authorized(&response.user.email, ui);
-            return Ok(token.into());
+            return Ok(Token {
+                token: token.into(),
+                is_existing: true,
+            });
+        }
+    }
+
+    // If the user is logging into Vercel, check for an existing `vc` token.
+    if login_url_configuration.contains("vercel.com") {
+        if let Some(vercel_config_dir) = turborepo_dirs::vercel_config_dir() {
+            let vercel_token_path = vercel_config_dir
+                .join(VERCEL_CONFIG_DIR)
+                .join(VERCEL_TOKEN_FILE);
+            #[derive(serde::Deserialize)]
+            struct VercelToken {
+                token: String,
+            }
+            let contents = std::fs::read_to_string(&vercel_token_path)?;
+            if let Ok(token) = serde_json::from_str::<VercelToken>(&contents) {
+                // extract the actual token out of the struct
+                let token = token.token;
+                if let Ok(response) = api_client.get_user(&token).await {
+                    println!(
+                        "{}",
+                        ui.apply(BOLD.apply_to("Existing Vercel token found!"))
+                    );
+                    ui::print_cli_authorized(&response.user.email, ui);
+                    return Ok(Token {
+                        token,
+                        is_existing: true,
+                    });
+                }
```

Comment:
This logic is duped from above, seems like a good candidate for extraction into a new function.